contrib/cirrus/pr-should-link-jira.t

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+#
+# tests for pr-should-link-jira.t
+#
+
+ME=$(basename $0)
+
+# Our test script queries github, for which we need token
+if [[ -z "$GITHUB_TOKEN" ]]; then
+    echo "$ME: Please set \$GITHUB_TOKEN" >&2
+    exit 1
+fi
+export CIRRUS_REPO_CLONE_TOKEN="$GITHUB_TOKEN"
+
+
+###############################################################################
+# BEGIN test cases
+#
+# Feel free to add as needed. Syntax is:
+#    <exit status> <commit message> <pr> # comments
+#
+# Where:
+#    exit status        is the expected exit status of the script
+#    commit message     commit message list
+#    dest branch        name of branch
+#    pr                 pr number
+#
+
+read -d '\n' msg_no_jira << EndOfText
+This is some text
+without a jira
+EndOfText
+
+read -d '\n' msg_invalid_jira << EndOfText
+This is some text
+without a jira
+Fixes #42
+More text...
+EndOfText
+
+read -d '\n' msg_jira << EndOfText
+This is some text
+with a jira
+Fixes https://issues.redhat.com/browse/RHEL-50507
+More text...
+EndOfText
+
+read -d '\n' msg_jira2 << EndOfText
+This is some text
+with a jira
+Fixes:  https://issues.redhat.com/browse/RHEL-50507
+More text...
+EndOfText
+
+read -d '\n' msg_multiple_jira << EndOfText
+This is some text
+with multiple jira lines
+Fixes  https://issues.redhat.com/browse/RHEL-50507
+More text...
+Fixes: https://issues.redhat.com/browse/RHEL-50506
+More text...
+EndOfText
+
+tests="
+0  8821 msg_no_jira main             not rhel branch, no link, should pass
```

Comment:
Why not use a real PR? Like #23514 and #23567? Even better, those plus #8890, so you can add this test:
```
0  8890 msg_no_jira  v1.6.4-rhel     no link, but PR has magic label, should pass
```
(I have taken the liberty of creating a `No Jira Link` label and applied it to this very ancient PR).

Reply:
I updated the table to use #8890 to test getting the tag. However, I'm not sure what you mean by using the others for the checks, as we're currently embedding the tested pr descriptions in the test itself, we're not getting it from github. The github PR is only used for checking the override tag atm.

Reply:
I made them 0000 for the non-queried ones, and #23514 for the ones we query that have no label. Does that look better?

Reply:
Better, thank you! LGTM.